lale/lib/sklearn/dummy_classifier.py

```diff
@@ -0,0 +1,135 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import sklearn
+import sklearn.dummy
+
+import lale.docstrings
+import lale.operators
+
+_hyperparams_schema = {
+    "allOf": [
+        {
+            "description": "This first object lists all constructor arguments with their types, but omits constraints for conditional hyperparameters.",
+            "type": "object",
+            "relevantToOptimizer": [],
+            "additionalProperties": False,
+            "required": [
+                "strategy",
+                "random_state",
+            ],
+            "properties": {
+                "strategy": {
+                    "description": """Strategy to use to generate predictions.
+- “stratified”: generates predictions by respecting the training set’s class distribution.
+- “most_frequent”: always predicts the most frequent label in the training set.
+- “prior”: always predicts the class that maximizes the class prior (like “most_frequent”) and predict_proba returns the class prior.
+- “uniform”: generates predictions uniformly at random.
+- “constant”: always predicts a constant label that is provided by the user. This is useful for metrics that evaluate a non-majority class""",
+                    "enum": [
+                        "stratified",
+                        "most_frequent",
+                        "prior",
+                        "uniform",
+                        "constant",
+                    ],
+                    "default": "prior",
+                },
+                "random_state": {
+                    "description": "Seed of pseudo-random number generator for shuffling data when solver == ‘sag’, ‘saga’ or ‘liblinear’.",
+                    "anyOf": [
+                        {
+                            "description": "RandomState used by np.random",
+                            "enum": [None],
+                        },
+                        {
+                            "description": "Use the provided random state, only affecting other users of that same random state instance.",
+                            "laleType": "numpy.random.RandomState",
+                        },
+                        {"description": "Explicit seed.", "type": "integer"},
+                    ],
+                    "default": None,
+                },
+                "constant": {
+                    "description": "The explicit constant as predicted by the “constant” strategy. This parameter is useful only for the “constant” strategy.",
+                    "side constraint": "",
```

Comment:
this line should be removed

Reply:
Previously it was decided to have a "side constraint" for "constant" schema. Should I remove it? If there is a property to add for "side constraint" I can add it now.

Reply:
sorry for the confusion.  a "side constraint", in our terminology, is something like this:
https://github.com/IBM/lale/blob/master/lale/lib/sklearn/logistic_regression.py#L320-L333
which adds additional constraints that related more than one hyperparameter.
However, I don't think it is needed for this PR.
the "side constraint" line above should definitely be removed, as it is probably what is currently breaking the tests.  I would guess that is due from the space in the key, although I did not investigate that closely.

Reply:
Thanks for the explanation. Now it's clear. I will remove that key and push again